src/components/Header/Header.jsx

```diff
@@ -1,10 +1,45 @@
-import React from "react";
-import './Header.scss'
+import "./Header.scss";
+import React, { useState, useEffect } from "react";
 
-const Header = ({title}) =>
+const Header = ({ title }) => {
+  const [data, setData] = useState([]);
+  const [city, setCity] = useState("");
+
+  function changeCity(){
+     setData(data);
+  }
+
+  useEffect(() => {
+    fetch(
+      `https://api.openweathermap.org/data/2.5/weather?q=liverpool&units=metric&appid=3b86046cce0de3be7cfa8369f4540b37`
+    )
+      .then((res) => res.json())
+      .then((data) => {
+        setData(data);
+      })
+      .catch((e) => console.log(e.message));
+  }, []);
```

Comment:
This is better but you are still saving the data in 2 different places (lines 9 and 18). You only need one. 
I think what you need to do is:
- `onChange` in the text input, save the user's input, just like you've done - don't change anything here it's perfect.
- on search button click, trigger the API call to fetch the new data (for the new city we just saved). How will you do this? useEffect is good when you want to react to a change in value, but in this particular case, you want to react to a button click instead. I recommend not using useEffect here. How about just a simple `getNewWeather` function?
- Replace the word 'liverpool' with `${city}` so the API call changes.

If you do all this correctly it should update the temperature for each new city (it works for me 😊)

Reply:
(If you're wondering when to use `useEffect`, it would be perfect if you didn't have a 'search' button. For example if you just had the text input but no button, then you would want to make a new API call _each time the user types a new letter_. In this case, you would still save 'city' `onChange` but you would add `city` as a dependency in the useEffect array, and the API call inside the useEffect would get triggered each time the value of `city` changes.
This is very inefficient because you would make too many calls to the API, so in this case it's best to move the API call to be triggered by a button click instead.)

Reply:
Thank you so much Lucy! It works now. I didn't realise that fetch API works without useEffect. :) Now I learned when to use useEffect.